webdriver-spec.html

```diff
@@ -7391,23 +7392,54 @@
         <li><p>Let <var>start y</var> be equal to the <code>y</code>
             property of <var>input state</var>.</li>
 
-        <li><p>Let <var>element id</var> be equal to
-            the <code>element</code> property of <var>action
+        <li><p>Let <var>origin</var> be equal to
+            the <code>origin</code> property of <var>action
             object</var>.</li>
 
-        <li><p>If <var>element id</var> is not undefined:
+        <li><p>Match on the value of <var>origin</var>:
+            <dl>
+              <dt><code>"viewport"</code></dt>
+              <dd><p>Let <var>x</var> equal <var>x offset</var> and
+                  <var>y</var> equal <var>y offset</var>.</dd>
 
-            <ol>
-              <li><p>Let <var>element</var> be equal to the result
-                  of <a>trying</a> to <var>get a known element</var> with
-                  argument <var>element</var>.</li>
+              <dt><code>"pointer"</code></dt>
+              <dd><p>Let <var>x</var> equal <var>start x</var>
+              + <var>x offset</var> and
+                  <var>y</var> equal <var>start y</var> + <var>y
+                  offset</var>.</dd>
 
-              <li><p>If <var>x</var> and <var>y</var> are <a>undefined</a>,
-                let <var>x</var> and <var>y</var> be the result of calculating the
-                <a>in-view centre point</a> of <var>element</var>.
-            </ol>
+              <dt>An object that <a>represents a web element</a></dt>
+              <dd>
+                <ol>
+                  <li><p>Let <var>element</var> be equal to the result
+                      of <a>trying</a> to <var>get a known
+                      element</var> with
+                      argument <var>origin</var>.</li>
+
+                  <li><p>Let <var>x element</var> and <var>y
+                  element</var> be the result of calculating the
+                    <a>in-view centre point</a>
+                    of <var>element</var>.</li>
+
+                  <li><p>Let <var>x</var> equal <var>x element</var>
+                      + <var>x offset</var>, and <var>y</var>
+                      equal <var>y element</var> + <var>y
+                      offset</var>.</li>
+                </ol>
+              </dd>
+            </dl>
         </li>
 
+        <li><p>If <var>x</var> is less than 0 or greater than the width
+            of the viewport in <a>CSS reference pixels</a>, then
```

Comment:
I'm happy to do that (it seems way more useful to me in general), but I think it should go via the list.